libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/NavigationSession.kt

```diff
@@ -123,3 +153,45 @@ sealed class NavigationSessionState {
         override val sessionId: String
     ) : NavigationSessionState()
 }
+
+sealed class NavigationSessionStateV2 {
+
+    /**
+     * Random session UUID.
+     * This is generated internally based on the current state within a trip session.
+     * I.e. will change when transitioning across states of a trip session. Empty when [Idle].
+     *
+     * Useful to use it in combination with the [MapboxHistoryRecorder].
+     *
+     * @see [TripSessionState]
+     */
+    abstract val sessionId: String
+
+    /**
+     * Idle state
+     */
+    object Idle : NavigationSessionStateV2() {
+        override val sessionId = ""
+    }
+
+    /**
+     * Free Drive state
+     */
+    data class FreeDrive internal constructor(
+        override val sessionId: String
+    ) : NavigationSessionStateV2()
+
+    /**
+     * Active Guidance state
+     */
+    data class ActiveGuidance internal constructor(
+        override val sessionId: String
+    ) : NavigationSessionStateV2()
+
+    /**
+     * Route preview state
+     */
+    data class RoutePreview internal constructor(
+        override val sessionId: String
+    ) : NavigationSessionStateV2()
+}
```

Comment:
Thinking it could be better to make this a string def. And what do you think about adding it to the `NavigationSessionState` instead of a `NavigationSessionState2`
```
    @StringDef(
        FREE_DRIVE,
        ACTIVE_GUIDANCE,
        ROUTE_PREVIEW,
```

Reply:
`ROUTE_PREVIEW` can be a type of the old `NavitationSessionState.FreeDrive`

Reply:
Here is a new state that is similar, it includes more states such as feedback and search. 

It also allows the customer to build custom states https://github.com/mapbox/mapbox-navigation-android/blob/77993f88fb7bf75bec4ab5c72dd61b37538267f3/libnavui-androidauto/src/main/java/com/mapbox/androidauto/screenmanager/MapboxScreen.kt#L91-L107

Reply:
Discussing, not putting any of these states here anyways. Resolving this thread